robot/impl/local_robot.go

```diff
@@ -434,6 +435,12 @@ func newWithResources(
 		r.packageManager = packages.NewDeferredPackageManager(
 			ctx,
 			func(ctx context.Context) (packagespb.PackageServiceClient, error) {
+				if conn != nil {
```

Comment:
Spoke offline, but these few lines are probably unnecessary if `AcquireConnection` gets the global app connection.